lib/maze/client/appium/bb_client.rb

```diff
@@ -6,7 +6,7 @@ def prepare_session
           config = Maze.config
           config.app = Maze::Client::BitBarClientUtils.upload_app config.access_key,
                                                                   config.app
-          if Maze.config.start_tunnel
+          if Maze.config.start_tunnel && !Maze.config.aws_public_ip
```

Comment:
This is duplicated 4 times and seems to be the new requirement for whether to start & stop the tunnel

Can we add a method for this, e.g. `Maze::Client::BitBarClientUtils.use_local_tunnel?`

Reply:
This could also be:

```suggestion
          if Maze::Client::BitBarClientUtils.use_local_tunnel?
            @started_local_tunnel = true
            # ...
```

Then `stop_session` can just check for `@started_local_tunnel`